### PR TITLE
closes #102, initial commit on curriculum creation and file extensions

### LINKED ISSUE #102: #1072
Copy repo into a new curricula owned by user

### PR DESCRIPTION
**Pull request by kdorer**

### CODE REVIEW

File: app/controllers/curricula_controller.rb [hunk 1]
```diff
@@ -45,11 +50,17 @@ def create
       @curricula = Curricula.new(curricula_params)
       @user = User.find(current_user.id)
       @curricula.users << @user
-      @curricula.creator = @user
+      @curricula.creator_id = @user.id
```
Comment (bammons) [inline app/controllers/curricula_controller.rb]: This is the incorrect way to do this. The user object should be assigned not the id. Just assigning the id doesn't create the user creator relationship

File: app/controllers/curricula_controller.rb [hunk 2]
```diff
@@ -63,10 +74,28 @@ def edit
     end
   end
 
-  def clone
-    @user = User.find_by_username(params[:username])
-    @curriculum = Curricula.where('creator_id=? AND cur_name=?', @user.id, params[:curriculum_name]).first
-    @path = Rails.root + @curriculum.path
+  def fork
+    @forked = Curricula.find_by_id(params[:id])
+    @creator = User.find(@forked.creator_id)
```
Comment (bammons) [inline app/controllers/curricula_controller.rb]: This query would be unnecessary if the creator assignment above was left as was. You could just get the creator by going @creator = @forked.creator.id

File: app/controllers/curricula_controller.rb [hunk 3]
```diff
@@ -63,10 +74,28 @@ def edit
...
+
+    @fork = Curricula.new
+    @fork.cur_name = @forked.cur_name
+    @fork.cur_description = @forked.cur_description
+    @fork.users << current_user
+    @fork.creator_id = current_user.id
```
Comment (bammons) [inline app/controllers/curricula_controller.rb]: similar changes need to happen here

File: app/controllers/curricula_controller.rb (hunk 1 above)
Comment (kdorer) [inline app/controllers/curricula_controller.rb]: I don't believe that I actually changed that, my version of develop that I branched from already contained that code.

Comment (bammons) [inline app/controllers/curricula_controller.rb]: Go ahead and merge develop into your branch and make fixes

**Comment by bammons** [comment]
:+1:

**Pull request merged**